app/src/main/java/de/tum/in/tumcampusapp/widgets/MVVWidget.java

```diff
@@ -0,0 +1,146 @@
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.widget.RemoteViews;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.managers.TransportManager;
+
+/**
+ * Implementation of App Widget functionality.
+ * App Widget Configuration implemented in {@link MVVWidgetConfigureActivity MVVWidgetConfigureActivity}
+ */
+public class MVVWidget extends AppWidgetProvider {
+
+    private static final String BROADCAST_ALARM_NAME = "de.tum.in.newtumcampus.intent.action.BROADCAST_MVVWIDGET_ALARM";
+    public static final String EXTRA_STATION_ID = "de.tum.in.newtumcampus.intent.action.MVV_WIDGET_EXTRA_STATION_ID";
+    private static boolean alarmIsSet = false;
+
+    @Override
+    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
+        for (int appWidgetId : appWidgetIds) {
+            updateAppWidget(context, appWidgetManager, appWidgetId);
+        }
+        setAlarm(context);
+        super.onUpdate(context, appWidgetManager, appWidgetIds);
+    }
+
+    @Override
+    public void onDeleted(Context context, int[] appWidgetIds) {
+        // When the user deletes the widget, delete the associated setting from the database.
+        TransportManager transportManager = new TransportManager(context);
+        for (int appWidgetId : appWidgetIds) {
+            transportManager.deleteWidget(appWidgetId);
+        }
+        super.onDeleted(context, appWidgetIds);
+    }
+
+    @Override
+    public void onEnabled(Context context) {
+        // Enter relevant functionality for when the first widget is created
+        setAlarm(context);
+    }
+
+    @Override
+    public void onDisabled(Context context) {
+        // Cancel alarm as the last widget has been removed
+        Intent intent = new Intent(context, MVVWidget.class);
+        PendingIntent sender = PendingIntent.getBroadcast(context, 0, intent, 0);
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        alarmManager.cancel(sender);
+        alarmIsSet = false;
+        super.onDisabled(context);
+    }
+
+    /**
+     * If no alarm is running yet a new repeating alarm is started
+     */
+    private static void setAlarm(Context context) {
+        if (alarmIsSet) return;
+        AlarmManager am = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        Intent intent = new Intent(context, MVVWidget.class);
+        intent.setAction(BROADCAST_ALARM_NAME);
+        PendingIntent pi = PendingIntent.getBroadcast(context, 0, intent, 0);
+        //After after 30 or 60 (android may delay) seconds
+        am.setRepeating(AlarmManager.RTC, System.currentTimeMillis(), 30000, pi);
```

Comment:
Thanks for that detailed explanation!
I just think that a named constant would be better readable than an inline constant
```java
private static final int UPDATE_DELAY = 30 * 1000;
```